app/controllers/gift_requests_controller.rb

```diff
@@ -40,14 +41,31 @@ def destroy_multiple
     end
   end
 
+  def catalog
+    @back_route = events_path
+
+    @event = Event.find(params[:id])
+    @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc).page(params[:page] || 1)
+  end
+
   private
 
   def gift_request_params
     params.require(:gift_request).permit(:id, :name, :recipient, :description, :link, :gender, :age)
   end
 
+  def require_therapist
+    unless current_user.try(:therapist?)
+      redirect_to root_path, alert: "You are not authorized to access that page."
```

Comment:
UX flow nit - might it be better to redirect the user back to the previous page if they get a 403, rather than back to the root path?